### PR TITLE
Retry on skopeo subcommands

### PR DESCRIPTION
**Pull request by QiWang19**
Applies retry on more skopeo subcommands.
Related issue: https://github.com/containers/skopeo/issues/918

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@mtrmac @vrothberg PTAL

**Comment by rhatdan** [comment]
@mtrmac @vrothberg Could we get this reviewed, I think it might be time for a new skopeo release?

**Comment by TomSweeneyRedHat** [comment]
Needs a rebase @QiWang19

**Comment by vrothberg** [comment]
> @mtrmac @vrothberg Could we get this reviewed, I think it might be time for a new skopeo release?

I won't manage today but on Monday :+1:

File: cmd/skopeo/list_tags.go [hunk 1]
```diff
@@ -113,13 +117,20 @@ func (opts *tagsOptions) run(args []string, stdout io.Writer) (retErr error) {
 	}
 
 	// Do transport-specific parsing and validation to get an image reference
-	imgRef, err := parseDockerRepositoryReference(args[0])
-	if err != nil {
+	var imgRef types.ImageReference
+	if err = retryIfNecessary(ctx, func() error {
+		imgRef, err = parseDockerRepositoryReference(args[0])
+		return err
+	}, opts.retryOpts); err != nil {
```
Comment (mtrmac) [inline cmd/skopeo/list_tags.go:124]: This is not necessary; `parseDockerRepositoryReference` is happens purely in local memory.

File: cmd/skopeo/sync.go [hunk 2]
```diff
@@ -509,9 +513,15 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) error {
 		return err
 	}
 
+	ctx, cancel := opts.global.commandTimeoutContext()
+	defer cancel()
+
```
Comment (mtrmac) [inline cmd/skopeo/sync.go:518]: A good find :)

File: cmd/skopeo/sync.go [hunk 3]
```diff
@@ -509,9 +513,15 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) error {
...
 	sourceArg := args[0]
-	srcRepoList, err := imagesToCopy(sourceArg, opts.source, sourceCtx)
-	if err != nil {
+	var srcRepoList []repoDescriptor
+	if err = retryIfNecessary(ctx, func() error {
+		srcRepoList, err = imagesToCopy(sourceArg, opts.source, sourceCtx)
```
Comment (mtrmac) [inline cmd/skopeo/sync.go:522]: (This applies a single `retryOpts.maxRetry` to the whole call, even if it contacts many registries and checks many repositories. I’m completely undecided whether it’s what the users expect; for now, this works well enough and gets us to 100% retry coverage.)

**Pull request merged**